### PR TITLE
Merge GitLab provenance generation into latest

### LINKED ISSUE #6373: (libnpmpublish) GitLab CI provenance
I'm interested in contributing provenance generation for GitLab CI, similar to existing GitHub Actions provenance support - https://github.com/npm/cli/blob/latest/workspaces/libnpmpublish/lib/provenance.js.

Initial GitLab OIDC support has been added to sigstore-js: https://github.com/sigstore/sigstore-js/pull/394, so all we should need to do here is add provenance generation when running in a GitLab environment.

GitLab issue: https://gitlab.com/gitlab-com/Product/-/issues/5632

cc @marshall007 @bdehamer

### PR DESCRIPTION
**Pull request by wlynch**
This merges https://github.com/npm/cli/pull/6375 into latest.

We were waiting on a working example in the npm UI, which we now have with https://www.npmjs.com/package/@ps-testing/gitlab-npm-provenance#provenance

There are some additional Fulcio claim changes being discussed in https://github.com/sigstore/fulcio/pull/1206, but this shouldn't affect anything with npm cli.

/cc @feelepxyz 

## References

Fixes https://github.com/npm/cli/issues/6373

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -15828,6 +15828,7 @@
       "license": "ISC",
       "dependencies": {
         "ci-info": "^3.6.1",
+        "libnpmpublish": "file:",
```
Comment (feelepxyz) [inline package-lock.json:15831]: What's going on here?

File: workspaces/libnpmpublish/lib/provenance.js [hunk 2]
```diff
@@ -1,69 +1,204 @@
...
 
-const BUILDER_ID = 'https://github.com/actions/runner'
-const BUILD_TYPE_PREFIX = 'https://github.com/npm/cli/gha'
-const BUILD_TYPE_VERSION = 'v2'
+const GITHUB_BUILDER_ID = 'https://github.com/actions/runner'
+const GITHUB_BUILD_TYPE_PREFIX = 'https://github.com/npm/cli/gha'
+const GITHUB_BUILD_TYPE_VERSION = 'v2'
+
+const GITLAB_BUILD_TYPE_PREFIX = 'https://github.com/npm/cli/gitlab'
+const GITLAB_BUILD_TYPE_VERSION = 'v0alpha1'
```
Comment (feelepxyz) [inline workspaces/libnpmpublish/lib/provenance.js:14]: Any point bumping this to something like `v0beta1`?

**Comment by feelepxyz** [comment]
> We were waiting on a working example in the npm UI, which we now have with https://www.npmjs.com/package/@ps-testing/gitlab-npm-provenance#provenance

@wlynch thanks for opening this up! 

I would like to hold off merging this until we've fixed up some of the links in the UI. Should get this done this week.

**Comment by wraithgar** [comment]
I think the PR that we want to land is the actual `provenance` branch itself.  If there are changes still to be made (i.e. `GITLAB_BUILD_TYPE_VERSION`)  please submit a PR to that branch.

**Comment by wraithgar** [comment]
https://github.com/npm/cli/pull/6526

**Pull request closed without merging**